loopchain/channel/channel_inner_service.py

```diff
@@ -460,11 +462,15 @@ def complain_leader(self, complained_leader_id, new_leader_id, block_height, pee
             self._channel_service.stop_leader_complain_timer()
             if next_new_leader == ChannelProperty().peer_id:
                 # Turn to Leader and Send Leader Complain Block
-                util.logger.notice(f"No I'm your father....")
+                util.logger.spam(f"No I'm your father....")
                 self._channel_service.state_machine.turn_to_leader()
             else:
-                util.logger.notice(f"I'm your Jedi.")
+                util.logger.spam(f"I'm your Jedi.")
+                # TODO check new leader is alive.
+                # if not
+                #     self._channel_service.start_leader_complain_timer()
                 self._channel_service.state_machine.turn_to_peer()
+            block_manager.epoch.prev_leader_id = next_new_leader
```

Comment:
`next_new_leader` is a result of `block_manager.epoch.complain_result()`
And it will be boolean value.

Reply:
epoch.complain will return next leader's peer_id